### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by glennferrie**
Just switching 'can' to 'must' in the opening statement.  It implies that it is an imperative to implement all of the methods.  It also is representative of what is documented here: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/keywords/interface

## Summary

same explanation as above.  simply changing the wording. 

Fixes #Issue_Number (if available)

### CODE REVIEW

**Review by adisakshya — COMMENTED** [review]
You can always partially implement an interface using abstract class, or throw ```NotImplementedException``` for the methods that you don't want to implement.

So I believe it is good to say that ```a class can implement the functionalities``` rather than saying ```a class must implement the functionalities```.

File: docs/csharp/programming-guide/interfaces/index.md [hunk 1]
```diff
@@ -9,7 +9,7 @@ ms.assetid: 2feda177-ce11-432d-81b4-d50f5f35fd37
 ---
 # Interfaces (C# Programming Guide)
 
-An interface contains definitions for a group of related functionalities that a [class](../../language-reference/keywords/class.md) or a [struct](../../language-reference/keywords/struct.md) can implement.
+An interface contains definitions for a group of related functionalities that a [class](../../language-reference/keywords/class.md) or a [struct](../../language-reference/keywords/struct.md) must implement.
```
Comment (BillWagner) [inline docs/csharp/programming-guide/interfaces/index.md]: This wording gives a bit of freedom on the "must":

```suggestion
An interface contains definitions for a group of related functionalities that a non-abstract [class](../../language-reference/keywords/class.md) or a [struct](../../language-reference/keywords/struct.md) must implement.
```

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for spotting this and suggesting better wording @glennferrie 

@adisakshya raises a valid point that abstract classes can declare support for an interface without defining those methods. 

See my suggestion to wiggle through those distinctions, which are covered more thoroughly in this section.

After accepting that suggestion, this is ready to merge.

File: docs/csharp/programming-guide/interfaces/index.md (hunk 1 above)
Comment (adisakshya) [reply docs/csharp/programming-guide/interfaces/index.md]: This seems more accurate.

**Review by BillWagner — APPROVED** [review]

**Pull request merged**